### PR TITLE
AN-835 Feature/exoplayer error detail

### PR DESCRIPTION
**Pull request by mschoenlaub**
### Work
- Fixes: https://bitmovin.atlassian.net/browse/AN-835
- Description: Refactored the error mapping functionality. Fix leakage of information into error message (moved this to error data). Also, a part of the stack trace is now logged, so that we can see which are the "UNKNOWN" esceptions.

### Checklist

- [x] Updated CHANGELOG.md

### CODE REVIEW

File: collector/src/main/java/com/bitmovin/analytics/data/ErrorCode.kt [hunk 1]
```diff
@@ -18,33 +18,15 @@
...
-    }
-
     @Override
-    public String toString() {
-        return errorCode + ": " + description;
+    override fun toString() : String {
+        return errorCode.toString() + ": " + description;
     }
+
+    data class ErrorData(val msg: String, val details: Array<String> = emptyArray())
```
Comment (Tigraine) [inline collector/src/main/java/com/bitmovin/analytics/data/ErrorCode.kt]: Can this be pushed one level higher? Reads a bit "choppy" `ErrorCode.ErrorData`?

File: collector-exoplayer/src/main/java/com/bitmovin/analytics/exoplayer/ExceptionMapper.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                errorCode.errorData = ErrorCode.ErrorData(error.message
+                        ?: "", error.topOfStacktrace)
+            }
+        }
+        return errorCode
+    }
+
+    override fun map(throwable: Throwable): ErrorCode {
+        return when (throwable) {
+            is ExoPlaybackException -> map(throwable)
```
Comment (peterpf) [inline collector-exoplayer/src/main/java/com/bitmovin/analytics/exoplayer/ExceptionMapper.kt]: how do i know which `map` will be called? Please make it more readable

File: collector-exoplayer/src/test/java/com/bitmovin/analytics/error/ErrorCodeTest.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.bitmovin.analytics.error
```
Comment (peterpf) [inline collector-exoplayer/src/test/java/com/bitmovin/analytics/error/ErrorCodeTest.kt]: Those tests don't check the errorCode :)

File: collector/src/test/java/com/bitmovin/analytics/error/ErrorDataTest.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.bitmovin.analytics.error
+
+import com.bitmovin.analytics.data.ErrorData
+import com.bitmovin.analytics.utils.DataSerializer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class UtilTest {
```
Comment (peterpf) [inline collector/src/test/java/com/bitmovin/analytics/error/ErrorDataTest.kt]:
```suggestion
class ErrorDataTest {
```
This is the same file as `collector-exoplayer/src/test/java/com/bitmovin/analytics/error/ErrorDataTest.kt` except that the className is different.
Is it necessary to duplicate those?

Comment (mschoenlaub) [reply collector/src/test/java/com/bitmovin/analytics/error/ErrorDataTest.kt]: Nope, was a leftover. thx for pointing out

**Pull request merged**